Chapter02/recipe-0011/README.md

```diff
@@ -0,0 +1,100 @@
+# Discovering the operating system
+
+CMake is a set of cross-platform tools. Nevertheless it can be very useful to
+know on which operating system (OS) the configure or build step is executed:
+either to tweak CMake code for a particular OS, to enable conditional
+compilation depending on the OS or to use compiler-specific extensions if
+available. In this recipe we will demonstrate how to use CMake to detect the
+OS.
+
+
+## Ingredients
+
+- A C++ compiler
+- CMake (any version higher than 3.0)
+
+
+## How to do it
+
+For this example we will use the following `CMakeLists.txt` together with a
+`hello-world.cpp` that we can reuse from recipe X (ADD LINK) although the C++
+source code does not matter for the essence of this recipe and you can even
+leave it out and comment out the last line containing
+`add_executable(hello-world hello-world.cpp)`.
+
+```cmake
+# set minimum cmake version
+cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+
+# project name and language
+project(recipe-0011 CXX)
+
+# detect operating system
+message(STATUS "We are on a ${CMAKE_SYSTEM_NAME} system")
+
+# print custom message depending on the operating system
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
```

Comment:
I think in CMake 3.0 you don't need to dereference variables in `if` statements anymore. See the rules of precedence for evaluation of CMake variables on the their docs.

Reply:
OK but I always found that confusing. In other languages variables have consistent referencing. Here the referencing changes on the context. But if without "$" is closer to the CMake doc, then I shall adopt it.

Reply:
Read up more on this - it is really complicated and confusing but will probably go for the naked variable name.

Reply:
Yes, that was my strategy in chapter 1. I wanted to explain more, but it really gets too complicated very quickly.